backend/sirius-web-compatibility/src/main/java/org/eclipse/sirius/web/compat/diagrams/EdgeMappingConverter.java

```diff
@@ -210,7 +210,11 @@ public EdgeDescription convert(EdgeMapping edgeMapping) {
         if (edgeMapping.isUseDomainElement()) {
             String semanticCandidatesExpression = edgeMapping.getSemanticCandidatesExpression();
             String preconditionExpression = edgeMapping.getPreconditionExpression();
-            semanticElementsProvider = new SemanticCandidatesProvider(this.interpreter, EOBJECT, semanticCandidatesExpression, preconditionExpression);
+            String domainClass = edgeMapping.getDomainClass();
```

Comment:
`DomainClassPredicate` (which is used underneath) already handles the case where the domain class is blank as equivalent to `EObject`. It just does not accept `null`. Proposal
```java
String domainClass = Optional.ofNullable(edgeMapping.getDomainClass()).orElse("");
```
and let `DomainClassPredicate` treat "" as "anything".